beam/core/src/main/java/cz/o2/proxima/beam/direct/io/BlockingQueueLogObserver.java

```diff
@@ -302,11 +317,16 @@ void stop() {
   }
 
   void stop(boolean nack) {
+    nackAllIncoming = nack;
     stopped.set(true);
```

Comment:
this is confusing, `stopped` flag is atomic and `nackAllIcoming` is not 🤔 why is the reason for this?

Reply:
after closer look, stooped can be replaced with `volatile` as we're not using any atomic methods (eg. `getAndSet()`)

Reply:
Yes, we can probably do that.